### ISSUE TITLE
GDPR compliance

### ISSUE THREAD

**Issue by samip5** [goauthentik/authentik#1551]
**Describe your question/**
Trying to figure out what are the requirements for GDPR complience, as an user needs to be presented/prompted for explicit concent to process their information.

**Relevant infos**
Public Matrix server, which is going to use Authentik as the user database/registry

This should probably also need an way to mark the user to have given their consent (and when), but they also need a way to withdraw it, even if that means deleting their account as a result.

**Comment by olmari** [comment]
Our Privacy policy statement (tietosuojaseloste) contains (when done 100%) all the relevant info already, hence the traditional tickbox "I have read and accept the privacy policy statement" / "Olen lukenut ja hyväksyn tietosuojaselosteen" or to that effect is very enough regardin GDPR-part, separately if needed some "I agree the terms" if we want to have some actual terms to be agreed to (service provided as-is, etc)

**Comment by samip5** [comment]
...

**Comment by BeryJu** [comment]
The main thing needed for this is a way for users to request all their data, maybe with Admin approval, the rest should mostly be doable with flows and custom stages, no?

**Comment by samip5** [comment]
Correct, but I think the back channel log-out is also needed for this to invalidate users logins to Synapse for example?

**Comment by BeryJu** [comment]
does synapse support back-channel logout? If it does that would greatly help with implementing it since I have found very few things that support it

**Comment by samip5** [comment]
> does synapse support back-channel logout? If it does that would greatly help with implementing it since I have found very few things that support it

Apperantly not, https://github.com/matrix-org/synapse/issues/11326

**Comment by BeryJu** [comment]
tbf authentik currently supports neither front-channel nor back-channel for OIDC, mostly because very few applications and libraries support it, so I haven't found a good reference implementation that I can test against/with (also I'm lazy)

**Comment by samip5** [comment]
So when does the OIDC authenicated sessions expire?

**Comment by samip5** [comment]
> So when does the OIDC authenicated sessions expire currently?
@BeryJu You're yet to answer the question?

**Comment by dominic-r** [comment]
> Needs functionality to export all of a users data (available for admins)

Resolved with https://github.com/goauthentik/authentik/pull/18088

**Comment by samip5** [comment]
> > Needs functionality to export all of a users data (available for admins)
> 
> Resolved with https://github.com/goauthentik/authentik/pull/18088

This needs to be on the non-enterprise version as well?

**Comment by dominic-r** [comment]
I've asked internally about this ^^ so I'm keeping the issue open.

**Comment by fheisler** [comment]
We already support custom flows and stages, which can be used for user consent + revocation. We also support backchannel logout now, to integrate with + end sessions in apps that support it.

Admin bulk user CSV exports are not related to any of this, but events exports could be. If users frequently submit requests for their personal data, an admin events exports feature could help with serving many requests (though this could also be done via API and then formatted).

This does not need to be non-enterprise to support running a public server with legal compliance requirements. I don't expect that homelab users will have GDPR constraints...

**Comment by samip5** [comment]
> I don't expect that homelab users will have GDPR constraints

The fact that it's homelab doesn't make GDPR less of a thing when there's potentially family members PII.

**Comment by dewi-tik** [comment]
GDPR doesn't apply to purely personal and domestic use. See here: https://gdpr.eu/companies-outside-of-europe/

> There are two important exceptions we should note here. First, the GDPR does not apply to “purely personal or household activity.” So if you’ve collected email addresses to organize a picnic with friends from work, rest assured you will not have to encrypt their contact info to comply with the GDPR (though you might want to anyway!). The GDPR only applies to organizations engaged in “professional or commercial activity.” So, if you’re collecting email addresses from friends to fundraise a side business project, then the GDPR may apply to you.

>The second exception is for organizations with fewer than 250 employees. Small- and medium-sized enterprises (SMEs) are not totally exempt from the GDPR, but the regulation does free them from record-keeping obligations in most cases (see Article 30.5).
